scaladoc/src/dotty/tools/scaladoc/renderers/HtmlRenderer.scala

```diff
@@ -63,10 +63,29 @@ class HtmlRenderer(rootPackage: Member, val members: Map[DRI, Member])(using ctx
           link = navigablePage.link.copy(dri = docsRootDRI),
           children = Nil
         ))
-      case Some(site) =>
-        site.orphanedTemplates.map(templateToPage(_, site))
+      case Some(siteContext) =>
+        (siteContext.orphanedTemplates :+ siteContext.indexTemplate()).map(templateToPage(_, siteContext))
+
+  /**
+   * Here we have to retrive index pages from hidden pages and replace fake index pages in navigable page tree.
+   */
+  private def getAllPages: Seq[Page] =
+
+    def traversePages(page: Page): (Page, Seq[Page]) =
+      val (newChildren, newPagesToRemove): (Seq[Page], Seq[Page]) = page.children.map(traversePages(_)).foldLeft((Seq[Page](), Seq[Page]())) {
```

Comment:
`(Seq[Page], Seq[Page])` can be turned into a case class to make things much easier